tsinfer/formats.py

```diff
@@ -1999,6 +1999,8 @@ def add_ancestor(self, start, end, time, focal_sites, haplotype):
             raise ValueError("haplotypes incorrect shape.")
         if np.any(haplotype >= self._num_alleles[start:end]):
             raise ValueError("haplotype values must be < num_alleles.")
+        # Ancestors must be added in decreasing time order, but buffering means we have
```

Comment:
I think it'd be more helpful to check at addition-time. We can do this in a similar way to what's done for sites - we keep the last site position in an instance variable, and check against that.

Reply:
Good call. Amended.